### DISCUSSION TITLE
MQTT Sending Power Delta

### DISCUSSION THREAD

**Discussion by Massimo65** [arendst/Tasmota#12509] [Q&A] [answered]
Good morning. Thanks to all of you for the wonderful work. I have installed and configured 2 Shelly EM. They work properly. I would need to receive in Home Assistant, through MQTT, the differential value of the power W / Kw between the current reading and the previous one. It's possible. Like the daily counter but showing the delta
Thank you

**Comment by barbudor** [comment, marked as answer]
I believe you are not interested by power delta (W) but by energy delta (in kW/h)
Tasmota reports the total from midnight as well as yesterday total.
~~If you are looking for the difference between reports, you will have to compute that yourself
It is possible to do that in a rule and publish it to another topic~~

EDIT : see sfromis answer below

Still handling differences is not safe. Should you miss a reading, you will loose information

**Reply by sfromis** [reply]
In the telemetry messages, the field "Period" shows the energy consumption (Wh) in the interval, like `"Period":30.370` which in this case matches the relatively fixed power consumption divided into periods.

To get decimals, I used `EnergyRes 3`

**Reply by barbudor** [reply]
hum, I don't seem to have this one on my plugs
```
18:35:11.793 MQT: tele/skt-minipc/SENSOR = {"Time":"2021-06-29T18:35:11+02:00","ENERGY":{"TotalStartTime":"2021-01-04T00:00:00","Total":9.592,"Yesterday":0.075,"Today":0.052,"Power":24.4,"ApparentPower":27.6,"ReactivePower":13.0,"Factor":0.88,"Voltage":245,"Current":0.113}} (retained)
```

But that's a message generated from a powerdelta... might be the reason

**Reply by sfromis** [reply]
Is that from TelePeriod? It will not appear for other reasons for publishing the payload, like PowerDelta.

**Reply by barbudor** [reply]
just what I added ;)

**Reply by sfromis** [reply]
Typing at the same time :-)

**Reply by barbudor** [reply]
Can see it now
```
18:39:27.877 MQT: tele/skt-minipc/SENSOR = {"Time":"2021-06-29T18:39:27+02:00","ENERGY":{"TotalStartTime":"2021-01-04T00:00:00","Total":9.593,"Yesterday":0.075,"Today":0.054,"Period":1.1,"Power":19.1,"ApparentPower":28.3,"ReactivePower":20.8,"Factor":0.68,"Voltage":244,"Current":0.116}} (retained)
```

**Reply by sfromis** [reply]
Yeah, I used EnergyRes as I prefer such values to have at least 3 significant digits, and then I can also cross-check numbers by summing up the Period values, in case of irregularities.
